### DISCUSSION TITLE
404 error

### DISCUSSION THREAD

**Discussion by esam95** [community/community#66815] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello,

I tried deploying my React project to github pages, however I'm only getting a blank page. It seems github pages is reading my original index.html and not the one in my build directory. I get a 404 error in the console because of this. How do i solve this?

https://esam95.github.io/Webbshop-livs/

**Comment by seyLu** [comment]
#### Hi @esam95, 

![image](https://github.com/community/community/assets/98249191/d7b589bf-c886-487b-be69-6d70fec0de6d)
The github pages site is properly built and deployed on `gh-pages` branch.

![image](https://github.com/community/community/assets/98249191/3614603f-1b70-4da1-a98e-6dcbe325c31c)
![Screenshot from 2023-09-12 14-12-39](https://github.com/community/community/assets/98249191/8d184440-a07d-414c-912d-7cef16d9f3b8)
![image](https://github.com/community/community/assets/98249191/e88b2275-5409-4385-bd67-a47d5a99ed21)
The issue is with the assets not configured properly.

You are using **Vite** and the way you fix this, is to add the following line in `vite.config.js`:
```diff
import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react'

// https://vitejs.dev/config/
export default defineConfig({
   plugins: [react()],
+  base: '/Webbshop-livs/'
})
```

You can read more at: [Vite Deploy Demo](https://github.com/sitek94/vite-deploy-demo)

**Reply by esam95** [reply]
Thanks for your help. I'm now getting a new error:
![image](https://github.com/community/community/assets/113107746/31267b05-48cd-4888-8d52-63cd1e64f870)

The error seems to be located here:
![image](https://github.com/community/community/assets/113107746/48047dbf-f3da-4a8b-a64e-da625002b57e)

**Comment by GhostOf0days** [comment, marked as answer]
If the website is dynamic, it is not supported on Github Pages. Instead, use Netlify, Heroku, Vercel, or Railway.

**Reply by esam95** [reply]
Thanks for the info, I didn't know that.

**Comment by edumixer** [comment]
Verdade, ele não é compatível se for um site dinâmico. Nos projetos em que eu faço usando o ViteJs, acabo usando na maioria das o Netlify pra hospedagem. Algumas vezes uso o Vercel que é bom tambem 👍🏻

**Comment by Uliana200407** [comment]
Maybe pages haven't  been deployed yet, you may try also to clear your browser, it helped me
